lib/flags.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+class Flags {
+  static from(...args) {
+    const values = new Map(args.map((v, i) => [v, i]));
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
+        if (args[0] instanceof Uint64) {
+          this.value = new Uint64(args[0]);
+        } else {
+          this.value = new Uint64(0);
+          args.forEach(arg => this.set(arg));
+        }
+      }
+
+      has(key) {
+        return values.has(key);
+      }
+
+      get(key) {
+        return values.has(key) ?
+          Uint64.shiftRight(this.value, values.get(key)).toUint32() & 1 : -1;
```

Comment:
Wouldn't it be better to return true/false (set/not-set) from this method?

Reply:
I agree but we should handle incorrect `key`s, though if we will throw errors in `set`, `unset` and `toggle` than I'm fine with throwing error here too. Also, it primarily depends on how `Flags` will be used and whether there will be used `Number` or `boolean`.

Reply:
Ok, let's make it return true/false and throw if `key` is not present.

/cc @belochub @nechaido